### PR TITLE
[RELAY][OP] conv2d_transpose

### PR DESCRIPTION
**Pull request by srkreddy1238**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

**Comment by tqchen** [comment]
please request reviews

**Comment by srkreddy1238** [comment]
@siju-samuel @junrushao1994 @MarisaKirisame welcome to review.

File: include/tvm/relay/attrs/nn.h [hunk 1]
```diff
@@ -77,6 +77,57 @@ struct SoftmaxAttrs : public tvm::AttrsNode<SoftmaxAttrs> {
   }
 };
 
+/*! \brief Attributes used in trasnposed convolution operator */
```
Comment (MarisaKirisame) [inline include/tvm/relay/attrs/nn.h]: trasnposed -> transposed

File: include/tvm/relay/attrs/nn.h [hunk 2]
```diff
@@ -77,6 +77,57 @@ struct SoftmaxAttrs : public tvm::AttrsNode<SoftmaxAttrs> {
...
+  std::string weight_layout;
+  DataType out_dtype;
+
+  TVM_DECLARE_ATTRS(Conv2DTransposeAttrs, "relay.attrs.Conv2DTransposeAttrs") {
+    TVM_ATTR_FIELD(channels)
+      .set_default(NullValue<IndexExpr>())
+      .describe("The dimensionality of the output space"
+                "i.e. the number of output channels in the convolution.");
+    TVM_ATTR_FIELD(kernel_size)
+      .describe("Specifies the dimensions of the convolution window.")
```
Comment (MarisaKirisame) [inline include/tvm/relay/attrs/nn.h]: Remove 'Specifies'?

File: include/tvm/relay/attrs/nn.h [hunk 3]
```diff
@@ -77,6 +77,57 @@ struct SoftmaxAttrs : public tvm::AttrsNode<SoftmaxAttrs> {
...
+      .set_default(NullValue<Array<IndexExpr> >());
+    TVM_ATTR_FIELD(strides).set_default(Array<IndexExpr>({1, 1}))
+      .describe("Specifies the strides of the convolution.");
```
Comment (MarisaKirisame) [inline include/tvm/relay/attrs/nn.h]: Remove 'Specifies'?

File: python/tvm/relay/op/nn/nn.py [hunk 4]
```diff
@@ -88,6 +88,62 @@ def conv2d(data,
...
+    data : relay.Expr
+        The input data to the operator.
+
+    weight : relay.Expr
+        The weight expressions.
+
+    strides : tuple of int, optional
+        The strides of convoltution.
+
+    padding : tuple of int, optional
```
Comment (MarisaKirisame) [inline python/tvm/relay/op/nn/nn.py]: Tuple[int, ...] as in mypy

File: src/relay/op/nn/convolution.cc [hunk 5]
```diff
@@ -154,5 +154,154 @@ with the layer input to produce a tensor of outputs.
...
+               2 * param->padding[0] + param->output_padding[0]);
+  oshape[3] = (param->strides[1] * (dshape_nchw[3] - 1) + dilated_ksize_x -
+               2 * param->padding[1] + param->output_padding[1]);
+
+  DataType out_dtype = param->out_dtype;
+  if (out_dtype.bits() == 0) {
+    out_dtype = data->dtype;
+  }
+  oshape = ConvertLayout(oshape, kNCHW, in_layout);
+  // assign output type
```
Comment (MarisaKirisame) [inline src/relay/op/nn/convolution.cc]: this line of comment is redundant

Comment (MarisaKirisame) [reply src/relay/op/nn/convolution.cc]: as it just describe the next line

**Comment by srkreddy1238** [comment]
cc @MarisaKirisame please take another look ?

**Pull request merged**